code/game/gamemodes/gameticker.dm

```diff
@@ -263,19 +263,18 @@ var/global/datum/controller/gameticker/ticker
 			M.locked_to = temp_buckle				//buckles the mob so it can't do anything
 			if(M.client)
 				M.client.screen += cinematic	//show every client the cinematic
-	else	//nuke kills everyone on z-level 1 to prevent "hurr-durr I survived"
+	else	//nuke kills everyone on the station to prevent "hurr-durr I survived"
 		for(var/mob/living/M in living_mob_list)
 			M.locked_to = temp_buckle
 			if(M.client)
 				M.client.screen += cinematic
 
-			switch(M.z)
-				if(0)	//inside a crate or something
-					var/turf/T = get_turf(M)
-					if(T && T.z==1)				//we don't use M.death(0) because it calls a for(/mob) loop and
-						M.nuke_act()
-				if(1)	//on a z-level 1 turf.
-					M.nuke_act()
+			if(!(M.z))	//inside a crate or something
+				var/turf/T = get_turf(M)
+				if(T && T.z==map.zMainStation)	//we don't use M.death(0) because it calls a for(/mob) loop and
+				M.nuke_act()
```

Comment:
It just works™
(Actually this is an amendment of a fix that worked but I guess I tabbed it incorrectly in a change afterwards)